Blocks/Persistence/test.cc

```diff
@@ -590,6 +684,125 @@ TEST(PersistenceLayer, FileSignatureExceptions) {
   }
 }
 
+TEST(PersistenceLayer, FileSafeVsUnsafeIterators) {
+  using namespace persistence_test;
+
+  using IMPL = current::persistence::File<StorableString>;
+
+  const auto namespace_name = current::ss::StreamNamespaceName("namespace", "entry_name");
+  const std::string persistence_file_name = current::FileSystem::JoinPath(FLAGS_persistence_test_tmpdir, "data");
+  const auto file_remover = current::FileSystem::ScopedRmFile(persistence_file_name);
+
+  current::reflection::StructSchema struct_schema;
+  struct_schema.AddType<StorableString>();
+  const std::string signature =
+      "#signature " + JSON(current::ss::StreamSignature(namespace_name, struct_schema.GetSchemaInfo())) + '\n';
+  std::string lines[] = {"{\"index\":0,\"us\":100}\t{\"s\":\"foo\"}\n",
```

Comment:
Perhaps `std::vector<std::string>`? Also, perhaps remove the trailing `\n`-s, and add them within `CombineFileContents`?